src/readers/opencv_reader.py

```diff
@@ -14,31 +14,43 @@
 # limitations under the License.
 
 import cv2
+from typing import Iterator, Dict
 
 from src.readers.abstract_reader import AbstractReader
 from src.utils.logging_manager import LoggingLevel
 from src.utils.logging_manager import LoggingManager
+from src.models.storage.frame import Frame
 
 
 class OpenCVReader(AbstractReader):
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, start_frame_id=0, *args, **kwargs):
         """
-            Reads video using OpenCV and yields frame data 
+            Reads video using OpenCV and yields frame data.
+            It will use the `start_frame_id` while annotating the
+            frames. The first frame will be annotated with `start_frame_id`
+            Attributes:
+                start_frame_id (int): id assigned to first read frame
+                    eg: start_frame_id=10, returned Iterator will be
+                    [{10, frame1}, {11, frame2} ...]
+                    It is different from offset. Offset defines where in video
+                    should we start reading. And start_frame_id defines the id
+                    we assign to first read frame.
          """
+        self._start_frame_id = start_frame_id
         super().__init__(*args, **kwargs)
 
-    def _read(self):
+    def _read(self) -> Iterator[Dict]:
         video = cv2.VideoCapture(self.file_url)
-        video_start = self.offset if self.offset else 0
-        video.set(cv2.CAP_PROP_POS_FRAMES, video_start)    
-        
+        video_offset = self.offset if self.offset else 0
+        video.set(cv2.CAP_PROP_POS_FRAMES, video_offset)
+
         LoggingManager().log("Reading frames", LoggingLevel.INFO)
 
         _, frame = video.read()
-        frame_ind = video_start - 1
+        frame_id = self._start_frame_id
 
         while frame is not None:
-            frame_ind += 1
-            yield frame
+            yield Frame(frame_id, frame, None).asdict()
```

Comment:
I feel using `Frame` for just conversion (asdict) is an overkill. We can instead create the dictionary directly.

Reply:
The intuition behind such an approach was to get rid of hardcoding the column names here.
But since we are removing Frames anyway so creating a dictionary directly.